### DISCUSSION TITLE
Could not find a package configuration file provided by "nlohmann_json"  while integrating the json library into Cmake project

### DISCUSSION THREAD

**Discussion by MdAnayat** [nlohmann/json#2317] [Q&A] [answered]
Dear All,
I have installed nlohmann json library using the below command on Ubuntu 18.04:
sudo apt update
sudo apt install nlohmann-json-dev
After that i have created a cmake project and my CMakeLists.txt file looks like below:

cmake_minimum_required (VERSION 3.5)

project(sign)

find_package(OpenCV REQUIRED)

find_package(nlohmann_json  REQUIRED)

include_directories(${OpenCV_INCLUDE_DIRS})

add_executable(sign_detect.exe sign_detect.cpp)

target_link_libraries(sign_detect.exe ${OpenCV_LIBS})

target_link_libraries(sign_detect.exe  nlohmann_json::nlohmann_json)

The result of this file gives me the below error:
CMake Error at CMakeLists.txt:7 (find_package):
  By not providing "Findnlohmann_json.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "nlohmann_json", but CMake did not find one.

  Could not find a package configuration file provided by "nlohmann_json"
  with any of the following names:

    nlohmann_jsonConfig.cmake
    nlohmann_json-config.cmake

  Add the installation prefix of "nlohmann_json" to CMAKE_PREFIX_PATH or set
  "nlohmann_json_DIR" to a directory containing one of the above files.  If
  "nlohmann_json" provides a separate development package or SDK, be sure it
  has been installed.
-- Configuring incomplete, errors occurred!

Any kind of help to solve this problem will be highly appreciated

Best Regards
Anayat

**Comment by nlohmann** [comment]
Maybe related: #2184

I am not maintaining any Ubuntu package, so I am afraid I cannot help.

**Reply by MdAnayat** [reply]
Thanx for your reply...
I have resolved the above problem by modifying my CMakeLists.txt file which is as follows:
cmake_minimum_required (VERSION 3.5)

project(sign)

find_package(OpenCV  REQUIRED)

find_package(nlohmann_json 2.1.1  REQUIRED PATHS /usr/lib/cmake)

include_directories(${OpenCV_INCLUDE_DIRS})

add_executable(sign_detect.exe sign_detect.cpp)

target_link_libraries(sign_detect.exe ${OpenCV_LIBS})

target_link_libraries(sign_detect.exe nlohmann_json)

But while trying to compile the cmake project using the make command i am getting the below error:

nvidia@nvidia-desktop:~/SEA-master/sea/app/services/sign_detect/build$ make
Scanning dependencies of target sign_detect.exe
[ 50%] Building CXX object CMakeFiles/sign_detect.exe.dir/sign_detect.cpp.o
/home/nvidia/SEA-master/sea/app/services/sign_detect/sign_detect.cpp: In function ‘int main(int, char**)’:
/home/nvidia/SEA-master/sea/app/services/sign_detect/sign_detect.cpp:484:9: **error: ‘using json = using json = class nlohmann::basic_json<> {aka class nlohmann::basic_json<>}’ has no member named ‘contains’
   if (j.contains(std::string{ blue_max })** 
         ^~~~~~~~
/home/nvidia/SEA-master/sea/app/services/sign_detect/sign_detect.cpp:485:9: **error: ‘using json = using json = class nlohmann::basic_json<> {aka class nlohmann::basic_json<>}’ has no member named ‘contains’
    && j.contains(std::string{ blue_min })**

Can you please help me to solve this problem???

**Reply by nlohmann** [reply]
Version 2.x is rather old, and does not have the contains() function yet.

**Comment by MdAnayat** [comment, marked as answer]
I have followed another approach to solve this problem. I have downloaded the latest json.hpp file from the repository and modified my CMakeLists.txt file as follows:

cmake_minimum_required (VERSION 3.5)

project(sign)

find_package(OpenCV  REQUIRED)

include_directories(${OpenCV_INCLUDE_DIRS})

include_directories(include)

add_executable(sign_detect sign_detect.cpp include/json.hpp)

target_link_libraries(sign_detect ${OpenCV_LIBS})

Now the code compilation is okay

**Reply by nlohmann** [reply]
Cool! Thanks for sharing!
